File: backend/src/main/java/es/udc/nutritfg/backend/model/services/FoodService.java
package es.udc.nutritfg.backend.model.services;

import java.util.List;

import es.udc.nutritfg.backend.model.common.exceptions.DuplicateInstanceException;
import es.udc.nutritfg.backend.model.common.exceptions.InstanceNotFoundException;
import es.udc.nutritfg.backend.model.entities.Dietician;
import es.udc.nutritfg.backend.model.entities.Food;
import es.udc.nutritfg.backend.model.entities.FoodGroup;
import es.udc.nutritfg.backend.rest.dtos.CountryDto;
import es.udc.nutritfg.backend.rest.dtos.FoodGroupDto;
import es.udc.nutritfg.backend.rest.dtos.MeasureUnitDto;
import es.udc.nutritfg.backend.rest.dtos.NutrientMeasureUnitsDto;

public interface FoodService {

	public Food addFood(Food food) throws DuplicateInstanceException;
	
	public List<FoodGroup> getFoodGroups();
	
	public Food findFoodById(Long foodId) throws InstanceNotFoundException;
	
	public Block<Food> findFoods(int page, int size);

	public Block<Food> searchFoods(Long groupFoodId, String keywords, int page, int size);
	
	public FoodGroup findFoodGroup(Long foodGroupId) throws InstanceNotFoundException;
	
	MeasureUnitDto getMeasureUnit();
	
	NutrientMeasureUnitsDto getNutrientMeasureUnits();
	
	public Block<Food> deleteFood(Long id, Long foodGroupId, String keywords, int page, int size) throws InstanceNotFoundException;
}

File: frontend/src/backend/foodService.js
import {config, appFetch, setServiceToken, getServiceToken, removeServiceToken, setReauthenticationCallback} from './appFetch';

export const addFood = (food, onSuccess, onErrors) =>
    appFetch('/food/addFood', config('POST', food),
        onSuccess, 
        onErrors);

export const getFoodGroups = (onSuccess) => 
        appFetch('/food/foodGroups', config('GET', null), onSuccess);

export const getNutrientMeasureUnits = (onSuccess) => 
        appFetch('/food/nutrientMeasureUnits', config('GET', null), onSuccess);

export const findFoods = ({page}, onSuccess) => {
        appFetch(`/food/listFoods?page=${page}`, config('GET', null), onSuccess);
}
        
export const findFoodsKeywords = ({foodGroupId,keywords, page}, onSuccess) => 
        appFetch(`/food/listSearchFoods?foodGroupId=${foodGroupId}&keywords=${keywords}&` +
                `page=${page}`, config('GET', null), onSuccess);
    
export const findFoodById = (id, onSuccess) => 
        appFetch(`/food/foodFind/${id}`, config('GET', null), onSuccess);

export const deleteFood = ({foodId, page, keywords,foodGroupId}, onSuccess) => {
            appFetch(`/food/${foodId}/removeFood?foodGroupId=${foodGroupId}&keywords=${keywords}&` +
                `page=${page}`, config('GET', null), onSuccess);
        }